woocommerce/woocommerce-gutenberg-products-block
# Update dependencies order

Discussed in https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1127#discussion_r343273546.

This PR:
* Updates all dependency imports so all externals and aliases are under `External dependencies` and relative paths are under `Internal dependencies`.
* To make my job easier, I adapted a ESLint rule from Gutenberg to detect this kind of errors. I included it in this PR, ideally we might want to publish it as an NPM package (maybe under `@woocommerce/scripts`, as discussed yesterday in Slack?), but for now I left it as a subfolder of `bin`.

### How to test the changes in this Pull Request:

1. Run `npm run build` and verify everything builds correctly.
2. Open a JS file and start playing with dependencies order, verify errors appear when dependencies are not under the correct section.
